From 0f219bb75f9ee3c7305b03deacb45785ee209ec4 Mon Sep 17 00:00:00 2001
From: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Date: Wed, 25 Mar 2015 03:08:16 +0530
Subject: [PATCH 1/2] be2net: Backport patch for dma_set_mask_and_coherent

If dma_set_mask_and_coherent is not defined, use pci_set_dma_mask
and pci_set_consistent_dma_mask

Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 2492c7c..4b68fe9 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -5151,6 +5151,23 @@ static int be_probe(struct pci_dev *pdev, const struct pci_device_id *pdev_id)
 			goto free_netdev;
 		}
 	}
+#else
+	status = pci_set_dma_mask(pdev, DMA_BIT_MASK(64));
+	if (status) {
+		status = pci_set_dma_mask(pdev, DMA_BIT_MASK(32));
+		if (status) {
+			dev_err(&pdev->dev, "Could not set PCI DMA Mask\n");
+			goto free_netdev;
+		}
+	}
+	status = pci_set_consistent_dma_mask(pdev, DMA_BIT_MASK(64));
+	if (status) {
+		status = pci_set_consistent_dma_mask(pdev, DMA_BIT_MASK(32));
+		if (status) {
+			dev_err(&pdev->dev, "Could not set PCI DMA consistent Mask\n");
+			goto free_netdev;
+		}
+	}
 #endif
 
 	status = pci_enable_pcie_error_reporting(pdev);
-- 
1.7.9.5

